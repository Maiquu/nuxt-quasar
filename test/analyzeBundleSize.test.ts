import { fileURLToPath } from 'node:url'
import { promisify } from 'node:util'
import { createHash } from 'node:crypto'
import { createReadStream } from 'node:fs'
import { finished } from 'node:stream/promises'
import { join } from 'node:path'
import { describe, expect, it } from 'vitest'
import { setup, useTestContext } from '@nuxt/test-utils'
import folderSizeCallback from 'fast-folder-size'

const folderSize = promisify(folderSizeCallback)

describe.skip('Analyze Bundle Size', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('./fixtures/basic', import.meta.url)),
    server: false,
  })

  const lockFile = fileURLToPath(new URL('../pnpm-lock.yaml', import.meta.url))
  const lockHash = await createFileHash(lockFile)

  it('client bundle size', async () => {
    const ctx = useTestContext()
    const { buildDir } = ctx.nuxt!.options
    const clientBundle = await folderSize(join(buildDir, 'dist/client')) || 0
    expect(roundKb(clientBundle)).toMatchSnapshot(lockHash)
  })

  it('nitro bundle size', async () => {
    const ctx = useTestContext()
    const { buildDir } = ctx.nuxt!.options
    const nitroBundle = await folderSize(join(buildDir, 'output')) || 0
    // Bundle size in rounded kb
    expect(roundKb(nitroBundle)).toMatchSnapshot(lockHash)
  })
})

async function createFileHash(path: string): Promise<string> {
  const hash = createHash('sha1')
  const stream = createReadStream(path)
  stream.pipe(hash)
  await finished(stream)
  return hash.digest('hex')
}

function roundKb(byteSize: number) {
  return Math.round(byteSize / 1000)
}
