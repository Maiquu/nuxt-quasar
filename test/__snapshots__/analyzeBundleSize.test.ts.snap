// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Analyze Bundle Size > client bundle size > 396413586aa87abee0183d93090cf378e4e2619b 1`] = `446`;

exports[`Analyze Bundle Size > client bundle size > cbfa0def0b3444acf63702c5eda5c13337434e51 1`] = `446`;

exports[`Analyze Bundle Size > client bundle size > e858649366ce7dd7d454485a2475021be23f3965 1`] = `441`;

exports[`Analyze Bundle Size > nitro bundle size > 396413586aa87abee0183d93090cf378e4e2619b 1`] = `2562`;

exports[`Analyze Bundle Size > nitro bundle size > cbfa0def0b3444acf63702c5eda5c13337434e51 1`] = `2539`;

exports[`Analyze Bundle Size > nitro bundle size > e858649366ce7dd7d454485a2475021be23f3965 1`] = `3063`;
